heron/metricscachemgr/tests/java/com/twitter/heron/metricscachemgr/metricscache/CacheCoreTest.java

```diff
@@ -0,0 +1,748 @@
+// Copyright 2016 Twitter. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+
+package com.twitter.heron.metricscachemgr.metricscache;
+
+import java.io.IOException;
+import java.nio.charset.Charset;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+
+import org.junit.AfterClass;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import com.twitter.heron.proto.tmaster.TopologyMaster;
+import com.twitter.heron.spi.metricsmgr.metrics.MetricsFilter;
+
+public class CacheCoreTest {
+  private static String debugFilePath =
+      "/tmp/" + CacheCoreTest.class.getSimpleName() + ".debug.txt";
+  private static Path file = null;
+  private static List<String> lines = null;
+
+  @BeforeClass
+  public static void before() throws Exception {
+    file = Paths.get(debugFilePath);
+    lines = new ArrayList<>();
+
+    lines.add(Paths.get(".").toAbsolutePath().normalize().toString());
+  }
+
+  @AfterClass
+  public static void after() throws IOException {
+    Files.write(file, lines, Charset.forName("UTF-8"));
+  }
+
+  @Test
+  public void testMetricTreeIndex() {
```

Comment:
These tests are way too large. Typically tests should be short and there should be many of them. Can you break this up into multiple tests, maybe using Mock objects if that helps? If you're testing a cache or an index typically you'd have tests to add, remove, update, etc. Also are there errors conditions that need to be tested?